src/detail-page/hooks/useTabData.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from 'react';
+import { Redirect, useHistory } from 'react-router-dom';
+import { backend, devOps, frontend, projectManager, qa, uxUi } from '../../data/detail-page/tabs/index';
+import { Category } from '../../data/detail-page/tabs/tabsInterfaces';
+import { ProfessionId } from '../../data/home-page/ITProfessions';
+import { JobsTab } from '../components/JobsTab';
+import { LearnTab } from '../components/LearnTab';
+import { ResourcesTab } from '../components/ResourcesTab';
+
+const getData = (professionId: ProfessionId) => {
+  switch (professionId) {
+    case 'frontend':
+      return frontend;
+    case 'backend':
+      return backend;
+    case 'devops':
+      return devOps;
+    case 'projectmanager':
+      return projectManager;
+    case 'qa':
+      return qa;
+    case 'uxui':
+      return uxUi;
+    default:
+      return [];
+  }
+};
+
+export const useTabData = (professionId: ProfessionId) => {
+  const [activeTab, setActiveTab] = useState(0);
+  const categories = getData(professionId);
+
+  const history = useHistory();
+
+  const getTabContent = (categoryName: String) => {
+    switch (categoryName) {
+      case 'jobs':
+        return <JobsTab professionId={professionId} />;
+      case 'learn':
+        return <LearnTab professionId={professionId} />;
+      case 'resources':
+        return <ResourcesTab professionId={professionId} />;
+      default:
+        return <Redirect to="/404" />;
+    }
+  };
```

Comment:
In `data/detail-page/tabs/tabsInterfaces` you created `CategoryId` type. Maybe we could use it here instead of `String` (`categoryName: CategoryId`). If we decide on this we should also change type in `TabPanel` component props (`getTabContent`)

Reply:
Done :)